x-pack/plugins/apm/public/components/alerting/rule_types/anomaly_rule_type/index.tsx

```diff
@@ -89,6 +98,19 @@ export function TransactionDurationAnomalyRuleType(props: Props) {
       onChange={(value) => setRuleParams('environment', value)}
       serviceName={params.serviceName}
     />,
+    <PopoverExpression
+      value={params.anomalyDetectorTypes.toString()}
```

Comment:
`params.anomalyDetectorTypes` should be of the type `ApmMlDetectorType[]`. We do not want to display these directly because the values are `txLatency, txThroughput, txFailureRate`. 

Instead, we should have a function similar to `getApmMlDetectorIndex`:

```ts
export function getApmMlDetectorLabel(type: ApmMlDetectorType) {
  switch(type) {
    case ApmMlDetectorType.txLatency:
      return i18n.translate('...', 'latency');
    // other cases
  }
}
```

Reply:
I was just answering this in previous comments, you were faster, I'll take a look tomorrow how we can do this